src/flownet/ert/_create_ert_setup.py

```diff
@@ -47,28 +47,34 @@ def _create_observation_file(
         Nothing
 
     """
-    num_training_dates = round(len(schedule.get_dates()) * training_set_fraction)
+    num_dates = len(schedule.get_dates())
+    num_training_dates = round(num_dates * training_set_fraction)
 
-    if yaml:
-        template = _TEMPLATE_ENVIRONMENT.get_template("observations.yamlobs.jinja2")
-        with open(obs_file, "w") as fh:
-            fh.write(
-                template.render(
-                    {
-                        "schedule": schedule,
-                        "error_config": config.flownet.data_source.simulation.vectors,
-                    }
-                )
-            )
-    else:
-        template = _TEMPLATE_ENVIRONMENT.get_template("observations.ertobs.jinja2")
-        with open(obs_file, "w") as fh:
+    export_settings = [
+        ["_complete", 1, num_dates],
+        ["_training", 1, num_training_dates],
+        ["_test", num_training_dates + 1, num_dates],
+    ]
+
+    prefix_name = os.path.splitext(obs_file)
```

Comment:
The `splitext()` function returns a tuple `(root, ext)`. You are only interested in the root of the file, so take out that part right away:

```suggestion
    file_root = os.path.splitext(obs_file)[0]
```